os/kernel/init/os_start.c

```diff
@@ -287,6 +287,20 @@ void os_start(void)
 {
 	int i;
 
+#ifdef CONFIG_APP_BINARY_SEPARATION
+	ASSERT(CONFIG_APP_HEAP_INDEX < CONFIG_KMM_NHEAPS);
```

Comment:
Let's make a build error instead of assert. Because they are predefined values, we can make a warning at build time. That's better usability I think.

Reply:
And for remain lines, we can move it into libelf_addrenv.c which uses CONFIG_APP_HEAP_INDEX.

Reply:
Done